docs/changes/tests/17051.feature.rst

```diff
@@ -0,0 +1,2 @@
+Add a public function ``astropy.tests.helper.assert_objets_equal``,
+that compares two ndarray objects at the attribute level.
```

Comment:
... that compares two objects by their shape and attributes, including those provided by ``.info``